src/lua/init.c

```diff
@@ -950,6 +950,19 @@ run_script_f(va_list ap)
 	bool aux_loop_is_run = false;
 	bool is_option_e_ran = false;
 
+	/*
+	 * Execute scripts or modules pointed by TT_PRELOAD
+	 * environment variable.
+	 */
+	lua_getfield(L, LUA_GLOBALSINDEX, "package");
+	lua_getfield(L, -1, "loaded");
+	lua_getfield(L, -1, "tarantool");
+	lua_getfield(L, -1, "_internal");
+	lua_getfield(L, -1, "run_preload");
+	if (luaT_call(L, 0, 0) != 0)
```

Comment:
(I was close to change the code to follow your suggestion, but grepped sources on `LUA_OK` and scratched my head.)

It is very usual for tarantool to check a return value against zero. Moreover, the [reference manual](https://www.lua.org/manual/5.1/manual.html#lua_pcall) says:

> The `lua_pcall` function returns 0 in case of success or one of the following error codes (defined in lua.h):
>
> `LUA_ERRRUN`: a runtime error.
> `LUA_ERRMEM`: memory allocation error. For such errors, Lua does not call the error handler function.
> `LUA_ERRERR`: error while running the error handler function.

Surprisingly, no `LUA_OK`. Across the whole manual.

And we have no checks against `LUA_OK` within the `src` directory.

I would stick with existing approach if you don't insist.

Reply:
OK, never mind.